File: UPCRideAndroid/app/src/main/java/com/example/upcride/activities/AdaptadorSolicitudes.java
package com.example.upcride.activities;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.upcride.R;

import java.util.List;

public class AdaptadorSolicitudes extends RecyclerView.Adapter<AdaptadorSolicitudes.ViewHolder>{
    public static class ViewHolder extends RecyclerView.ViewHolder{

        private TextView nombre, fecha, descripcion;
        public ImageView foto_perfil;

        public ViewHolder(View itemView){
            super(itemView);
            nombre = (TextView) itemView.findViewById(R.id.n_usuario);
            fecha = (TextView) itemView.findViewById(R.id.d_fecha);
            descripcion = (TextView) itemView.findViewById(R.id.t_descripcion);
            foto_perfil = (ImageView) itemView.findViewById(R.id.foto_perfil);
        }
    }

    public List<SolicitudModelo> solicitudLista;

    public AdaptadorSolicitudes(List<SolicitudModelo> solicitudListaLista){
        this.solicitudLista = solicitudListaLista;
    }

    @Override
    public AdaptadorSolicitudes.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType){
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_solicitud, parent, false);
        AdaptadorSolicitudes.ViewHolder viewHolder = new AdaptadorSolicitudes.ViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull AdaptadorSolicitudes.ViewHolder holder, int position) {
        holder.nombre.setText(solicitudLista.get(position).getNombre());
        holder.fecha.setText(solicitudLista.get(position).getFecha());
        holder.descripcion.setText(solicitudLista.get(position).getDescripcion());
        holder.foto_perfil.setImageResource(solicitudLista.get(position).getFoto_perfil());
    }

    @Override
    public int getItemCount() {
        return solicitudLista.size();
    }
}

File: UPCRideKotlin/app/src/main/java/com/example/upcridekotlin/activities/solicitudesactivity/AdaptadorSolicitudes.kt
package com.example.upcridekotlin.activities.solicitudesactivity

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.example.upcridekotlin.R
import com.example.upcridekotlin.model.SolicitudModelo

class AdaptadorSolicitudes(var solicitudLista: List<SolicitudModelo>) :
    RecyclerView.Adapter<AdaptadorSolicitudes.ViewHolder>() {
    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var nombre: TextView
        var fecha: TextView
        var descripcion: TextView
        var foto_perfil: ImageView

        init {
            nombre = itemView.findViewById<View>(R.id.n_usuario) as TextView
            fecha = itemView.findViewById<View>(R.id.d_fecha) as TextView
            descripcion = itemView.findViewById<View>(R.id.t_descripcion) as TextView
            foto_perfil = itemView.findViewById<View>(R.id.foto_perfil) as ImageView
        }
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.item_solicitud, parent, false)
        return ViewHolder(
            view
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.nombre.setText(solicitudLista[position].nombre)
        holder.fecha.setText(solicitudLista[position].fecha)
        holder.descripcion.setText(solicitudLista[position].descripcion)
        holder.foto_perfil.setImageResource(solicitudLista[position].foto_perfil)
    }

    override fun getItemCount(): Int {
        return solicitudLista.size
    }
}
